### ISSUE TITLE
draft.software to identify r-universe packages

### ISSUE THREAD

**Issue by iagomosqueira** [ices-tools-prod/TAF#14]
A number of packages used in ICES are now available via [r-universe](https://r-universe.dev), including [FLR](https://flr.r-universe.dev/packages) and [SAM](https://fishfollower.r-universe.dev/packages). This includes binary versions, so it greatly simplifies installation for windows users.

The `packageDescription` of packages installed from [r-universe](https://r-universe.dev) contain both information about the package repository and the corresponding git hash of the commit used to create it

```
> packageDescription('FLCore')
Package: FLCore
[...]
Repository: https://flr.r-universe.dev
RemoteUrl: https://github.com/flr/FLCore
RemoteRef: HEAD
RemoteSha: d7a991864677b113f57cbb4df8431cc5d709f608
[...]
```

would it possible for `draft.software` to pick up this information and set 'source' from the github url and hash?

The workflow I am trying to support is to (1) start work by installing from r-universe, (2) updating from there when a package has been updated, and (3) generate `SOFTWARE.bib` once the analysis is done to keep a record of the versions used.

**Comment by arni-magnusson** [comment]
From a maintainer perspective, the short answer is 'sounds good'.

The almost-as-short answer is that I have limited understanding and no experience with r-universe :)

A way forward could be to:
1. Demonstrate a worked example (or examples), perhaps on https://github.com/ices-taf-dev.
2. Draft some code that adds the desired functionality, perhaps inside the same repository on https://github.com/ices-taf-dev.
3. Continue discussion here.

**Comment by iagomosqueira** [comment]
I am submitting a [pull request](https://github.com/ices-tools-prod/TAF/pull/17) with a proposal. A new repotype 'Runiverse' is defined in ds.packages. It assembles the information the 'GitHub' repotype gathers (username/repository and folder, plus commit SHA key), from the fields in DESCRIPTION set by r-universe.dev. The output SOFTWARE.bib should be identical to the one generated from a `install_github`-installed package.

**Comment by arni-magnusson** [comment]
Thanks! Your commit e9fdb7f has been merged into the main branch.

With commit 5a082a5 I've extended the functionality to support subdirs, so `draft.software("stockassessment")` will do the right thing.
